app/src/main/java/com/hedvig/app/feature/genericauth/GenericAuthActivity.kt

```diff
@@ -20,7 +21,13 @@ class GenericAuthActivity : BaseActivity() {
 
         setContent {
             val viewState by model.viewState.collectAsState()
+            val events = model.eventsFlow.collectAsState(initial = null)
+
             HedvigTheme {
+                when (val value = events.value) {
+                    is GenericAuthViewModel.Event.SubmitEmailSuccess -> startOtpInputActivity(value)
+                }
```

Comment:
I think receiving one time events as state and doing the `startOtpInputActivity` in composition is dangerous as "composables should be side-effect free" https://developer.android.com/jetpack/compose/side-effects .
Since the event will be saved as state, if the composable recomposes for whatever reason (there's a lot of reasons why it potentially could even if it maybe doesn't right now) it will trigger this again.
We should be subscribing to the state outside of composition in some way. Maybe something like (I wrote this outside of the IDE not sure if it compiles but you get the idea):
```kotlin
LaunchedEffect(Unit) {
    model.eventsFlow // This could be named just events too, "...Flow" suffix is maybe not necessary
        .collect { event -> // Or use `collectLatest` if we want to drop events and not chain them
            when (val value = model.eventsFlow) {
                is GenericAuthViewModel.Event.SubmitEmailSuccess -> startOtpInputActivity(value)
            }
        }
}
```

Reply:
Thanks for clarifying!! I'll read up and fix 👍